gxa/src/test/java/uk/ac/ebi/atlas/experimentpage/differential/DifferentialPathwaysComparisonServiceTest.java

```diff
@@ -0,0 +1,247 @@
+    private String[] c1s10 = new String[]{"R-MMU-141430", "Apoptosis", "45", "0.535169358108118", "0.87881981279076"};
+    private String[] c1s11 = new String[]{"R-MMU-170822", "Apoptosis", "45", "0.535169358108118", "1"};
+
+    private String[] c2s1 = new String[]{"Term", "Accession", "Genes (tot)", "Stat (non-dir.) p", "p adj (non-dir.)"};
+    private String[] c2s2 = new String[]{"R-MMU-416482", "G alpha (12/13) signalling events", "33", "0.538397547287844", "0.0191488220970447"};
+    private String[] c2s3 = new String[]{"R-MMU-2132295", "Apoptosis", "16", "0.535169358108118", "0.0280433635326222"};
+    private String[] c2s4 = new String[]{"R-MMU-5620924", "Apoptosis", "23", "0.535169358108118", "0.0280433635326222"};
+    private String[] c2s5 = new String[]{"R-MMU-397014", "Muscle contraction", "10", "0.535169358108118", "0.0280433635326222"};
+    private String[] c2s6 = new String[]{"R-MMU-193648", "NRAGE signals death through JNK", "45", "0.535169358108118", "0.0280433635326222"};
+    private String[] c2s7 = new String[]{"R-MMU-373760", "Apoptosis", "45", "0.535169358108118", "0.0280433635326222"};
+    private String[] c2s8 = new String[]{"R-MMU-199418", "Negative regulation of the PI3K/AKT network", "45", "0.535169358108118", "0.0436946551249873"};
+    private String[] c2s9 = new String[]{"R-MMU-2029480", "Fcgamma receptor (FCGR) dependent phagocytosis", "45", "0.535169358108118", "0.0436946551249873"};
+    private String[] c2s10 = new String[]{"R-MMU-5654741", "Signaling by FGFR3", "45", "0.535169358108118", "0.0436946551249873"};
+    private String[] c2s11 = new String[]{"R-MMU-1483206", "Apoptosis", "45", "0.535169358108118", "0.0436946551249873"};
+
+
+    @Before
+    public void setUp() throws Exception {
+        MockDataFileHub mockDataFileHub = new MockDataFileHub();
+
+        String accession = "E-GEOD-54351";
+
+        mockDataFileHub.addReactomePathwaysFile(accession, c1.getId(),
+                Lists.newArrayList(c1s1, c1s2, c1s3, c1s4, c1s5, c1s6, c1s7, c1s8, c1s9, c1s10, c1s11));
+        mockDataFileHub.addReactomePathwaysFile(accession, c2.getId(),
+                Lists.newArrayList(c2s1, c2s2, c2s3, c2s4, c2s5, c2s6, c2s7, c2s8, c2s9, c2s10, c2s11));
+
+        this.subject = new DifferentialPathwaysComparisonService<>(mockDataFileHub, mockConfigurationTrader,
+                mockDifferentialRequestContextFactory, mockDifferentialProfilesHeatMap);
+
+        when(mockDifferentialExperiment.getAccession()).thenReturn(accession);
+        when(mockConfigurationTrader.getMicroarrayExperimentConfiguration(accession)).thenReturn(mockMicroarrayExperimentConfiguration);
+        when(mockConfigurationTrader.getMicroarrayExperimentConfiguration(accession).getContrasts()).thenReturn(Lists.newArrayList(c1,c2));
+
+        setUpDataForTest();
+    }
+
+    @Test
+    public void testGetPathwaysListPerComparison() throws Exception {
```

Comment:
This is a okay test - it will tell us when someone makes a change to the code and it'll be wrong - but it won't be easy to work with when something is failing. A better test explains why the code does what it does, and asserts various things about its behaviour.
E.g.
"only values with pValue less than 0.05 are taken into account".
"if there is no file, throw an exception"
"if there are many pathways to choose from return top ten pathways"

Reply:
I can add a failing test when there is no file or there is something wrong